components/automate-gateway/handler/iam/v2beta/teams/teams.go

```diff
@@ -9,12 +9,18 @@ import (
 	gwcommon "github.com/chef/automate/components/automate-gateway/api/iam/v2beta/common"
 	gwreq "github.com/chef/automate/components/automate-gateway/api/iam/v2beta/request"
 	gwres "github.com/chef/automate/components/automate-gateway/api/iam/v2beta/response"
+	"github.com/chef/automate/components/automate-gateway/gateway/middleware"
 )
 
 type Server struct {
 	client teams.TeamsV2Client
 }
 
+// asserts that we satisfy the correct interface here -- it's a safeguard
+var _ middleware.AuthContextReader = (*Server)(nil)
```

Comment:
🤔

Reply:
That is a true 🤔 for sure.

https://github.com/chef/automate/blob/918c76098729134b39e2126e6d036e25f2ee7973/components/automate-gateway/gateway/middleware/auth_interceptor.go#L154-L166

All requests get the [projects headers atm](https://github.com/chef/automate/blob/918c76098729134b39e2126e6d036e25f2ee7973/components/automate-gateway/gateway/middleware/auth_interceptor.go#L164). However, only handlers that implements the empty interface `AuthContextReader` get all the [other auth headers](https://github.com/chef/automate/blob/918c76098729134b39e2126e6d036e25f2ee7973/components/automate-gateway/gateway/middleware/auth_interceptor.go#L162) (including the subjects header which we now need for project diff calculation).

My TODO there says we should consider just dropping this and alway send the auth headers with every request since we are going to mostly need them in every service eventually, but that has performance impact so I wanted to discuss it more widely before making that change.

So for now, you'll need to add that line to the tokens handler so that the subjects header shows up.

Reply:
Should probably replace most of the places I say headers with context. Still in http think.

Reply:
That totally helps clarify, thanks!

Reply:
> My TODO there says we should consider just dropping this and alway send the auth headers with every request since we are going to mostly need them in every service eventually

If we need them, we need them. The performance impact can't hinder us -- but it may necessitate optimizations, or different approaches. For now, I'd lean towards either dropping this, or, if need be, making it an **opt-out** instead.